File: react_frontend/src/service/StudentService.js
import { api } from "../api/api";

// React service
export const studentList = () => {
  return api.get("/studentList");
};

export const studentById = (studentName) => {
  return api.get(`/studentById?name=${studentName}`);
};

export const studentByName = (studentId) => {
  return api.get(`/studentByName?student_id=${studentId}`);
};

export const addStudent = (data) => {
  return api.post("/addStudent", data);
};

export const deleteStudent = (studentId) => {
  return api.delete(`/deleteStudent/${studentId}`);
};

export const editStudent = (data) => {
  return api.put("/editStudent", data);
};

File: react_springboot/src/main/java/com/student_management/react_springboot/service/StudentService.java
package com.student_management.react_springboot.service;

import java.util.List;

import com.student_management.react_springboot.model.StudentModel;

public interface StudentService {
	
	public int addStudent(StudentModel studentModel);
	public List<StudentModel> studentList();
	public int editStudent (StudentModel studentModel);
	public int deleteStudent(int studentId);
	public StudentModel studentById(String studentName);
	public  StudentModel studentByName(int studentId);
}

